rustls/src/msgs/persist.rs

```diff
@@ -133,6 +133,41 @@ impl Tls13ClientSessionValue {
         }
     }
 
+    /// [`Codec::read()`] with an extra `suites` argument.
+    ///
+    /// We need a reference to the available cipher suites in order to check
+    /// that the cipher suite in the ticket is supported.
+    pub fn read(
+        r: &mut Reader,
+        mut suites: impl Iterator<Item = &'static Tls13CipherSuite>,
+    ) -> Result<Self, InvalidMessage> {
+        let suite = CipherSuite::read(r)?;
+        let suite = suites
+            .find(|s| s.common.suite == suite)
+            .ok_or(InvalidMessage::UnsupportedCipherSuite(suite))?;
+
+        Ok(Self {
+            suite,
+            age_add: u32::read(r)?,
+            max_early_data_size: u32::read(r)?,
+            common: ClientSessionCommon::read(r)?,
+            #[cfg(feature = "quic")]
+            quic_params: PayloadU16::read(r)?,
+        })
+    }
+
+    /// Inherent implementation of the [`Codec::get_encoding()`] method.
```

Comment:
tiny nit: it seems like this is more like an impl. of `Codec::encode` since it takes a `Vec<u8>` as an argument instead of constructing one on behalf of the caller ala `get_encoding`.

Reply:
Yup, copy/pasta error. Thanks for catching it!